dotCMS/src/main/java/com/dotcms/contenttype/business/StoryBlockAPIImpl.java

```diff
@@ -112,12 +113,18 @@ private boolean refreshStoryBlockMap(final Map<String, Object> contentMap) throw
                 final String inode = (String) dataMap.get(INODE_KEY);
                 final long languageId = ConversionUtils.toLong(dataMap.get(LANGUAGE_ID_KEY), ()-> APILocator.getLanguageAPI().getDefaultLanguage().getId());
                 if (UtilMethods.isSet(identifier) && UtilMethods.isSet(inode)) {
-                    final Optional<ContentletVersionInfo> versionInfo = APILocator.getVersionableAPI().getContentletVersionInfo(identifier, languageId);
-                    if (versionInfo.isPresent() && UtilMethods.isSet(versionInfo.get().getLiveInode())  &&
+
+                    if (!parentContentletIdentifier.equals(identifier)) { // if somebody adds a story block to itself, we don't want to refresh it
```

Comment:
Please null check `parentContentletIdentifier` before using

Reply:
I have switch the sentence since identifier has a check in advance, and something not null can be check with something null such as
```
if (UtilMethods.isSet(identifier) && UtilMethods.isSet(inode)) {

                    if (!identifier.equals(parentContentletIdentifier)) { // if somebody adds a story block to itself, we don't want to refresh it
                    
                    ```